### PR TITLE
Initial commit for the macOS TCC plugin

### PR DESCRIPTION
**Pull request by tomchop**
## One line description of pull request

Plugin for Apple's Transparency, Consent, and Control database.

## Description:

Lists entries in Apple's TCC, including

* Name of service
* App that requested access to the service
* Whether the request was granted or refused
* How many times the app requested access to the servcice
* Timestamp where the app last requested access to the service

**Related issue (if applicable):** fixes #<plaso issue number here>

## Checklist:
  - [x] Automated checks (Travis, Codecov, Codefactor )pass
  - [x] No new [new dependencies](https://plaso.readthedocs.io/en/latest/sources/developer/Adding-a-new-dependency.html) are required or l2tdevtools has been updated
  - [x] Reviewer assigned

### CODE REVIEW

File: plaso/formatters/macos_tcc.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  def __init__(self):
+    """Initializes an iMessage chat event format helper."""
+    super(MacOSTCCFormatter, self).__init__()
+    helper = interface.EnumerationEventFormatterHelper(
+        default='UNKNOWN', input_attribute='allowed',
+        output_attribute='allowed', values=self._ALLOWED)
+
+    self.helpers.append(helper)
+
```
Comment (joachimmetz) [inline plaso/formatters/macos_tcc.py:40]: nit: `+1` blank line (I've already added this so consider this as an informational comment)

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
+# -*- coding: utf-8 -*-
+"""Parser for Apple's Transaprency, Consent, Control (TCC) framework database.
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:2]: nit: Keep this a single line (I've changed it already)

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+from plaso.containers import time_events
+from plaso.parsers import sqlite
+from plaso.parsers.sqlite_plugins import interface
+
+
+class MacOSTCCEntry(events.EventData):
+  """macOS TCC event data.
+
+  Attributes:
+    allowed (bool): whether access to the service was allowed.
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:23]: Nit: sort these alphabetically (already changed)

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    client (str): name of the client requesting access to the service.
+    prompt_count (int): number of times an appplication prompted for access
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:25]: I assume this prompts the user? Let's mention that in the docstring.

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+        to a service.
+    service (str): name of the service.
+    timestamp (int): last time the application's privacy permissions were
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:28]: This should not be part of the event data but an event. If you want I can explain this more over chat.

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+        modified.
+  """
+
+  DATA_TYPE = 'macos:tcc_entry'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(MacOSTCCEntry, self).__init__(data_type=self.DATA_TYPE)
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:36]: nit: this fits on a single line (has been changed)

File: plaso/parsers/sqlite_plugins/macos_tcc.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+              indirect_object_code_identity BLOB,
+              flags INTEGER,
+              last_modified  INTEGER NOT NULL
+                  DEFAULT (CAST(strftime('%s','now') AS INTEGER)),
+            PRIMARY KEY
+                (service, client, client_type, indirect_object_identifier),
+            FOREIGN KEY (policy_id)
+            REFERENCES policies(id) ON DELETE CASCADE ON UPDATE CASCADE);""")}]
+
+  def ParseTCCEntry(self, parser_mediator, query, row, **unused_kwargs):
```
Comment (joachimmetz) [inline plaso/parsers/sqlite_plugins/macos_tcc.py:83]: nit: this fits on a single line (has been changed)

File: tests/parsers/sqlite_plugins/macos_tcc.py [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  def testProcess(self):
+    """Tests the Process function on a macOS TCC file."""
+    plugin = macos_tcc.MacOSTCCPlugin()
+    storage_writer = self._ParseDatabaseFileWithPlugin(
+        ['TCC-test.db'], plugin)
+
+    self.assertEqual(storage_writer.number_of_warnings, 0)
+    self.assertEqual(storage_writer.number_of_events, 21)
+
+    events = list(storage_writer.GetEvents())
```
Comment (joachimmetz) [inline tests/parsers/sqlite_plugins/macos_tcc.py:26]: Please use CheckEventValues also see https://github.com/log2timeline/plaso/blob/master/tests/parsers/winprefetch.py#L70 for an example

Comment (tomchop) [reply tests/parsers/sqlite_plugins/macos_tcc.py:26]: Done

File: plaso/parsers/sqlite_plugins/macos_tcc.py (hunk 5 above)
Comment (tomchop) [reply plaso/parsers/sqlite_plugins/macos_tcc.py]: Yup, got it. I removed the attribute from the class.

File: plaso/parsers/sqlite_plugins/macos_tcc.py (hunk 4 above)
Comment (tomchop) [reply plaso/parsers/sqlite_plugins/macos_tcc.py]: Fixed.

File: plaso/parsers/sqlite_plugins/macos_tcc.py (hunk 5 above)
Comment (joachimmetz) [reply plaso/parsers/sqlite_plugins/macos_tcc.py]: ack, thx

**Pull request merged**